index.js

```diff
@@ -3232,6 +3232,55 @@
   S.range =
   def('range', {}, [$.Integer, $.Integer, $.Array($.Integer)], range);
 
+  //# groupBy :: (a -> a -> Boolean) -> Array a -> Array (Array a)
+  //.
+  //. Splits its list argument into a list of lists of equal,
+  //. adjacent elements. The equality of the elements is tested with
+  //. a curried function provided by the user in the first argument.
+  //. The behavior of groupBy aims to be equal to Haskell's groupby,
+  //. rather than Ramda's groupBy.
+  //.
+  //. The behavior of groupBy can be surprising for functions that
+  //. aren't reflexive, transitive and symmetric
+  //. (see [equivalence][] relation).
+  //.
+  //. ```javascript
+  //. > S.groupBy(S.equals, [1, 1, 2, 3, 3, 4])
+  //. [[1, 1], [2], [3, 3], [4]]
+  //.
+  //. > S.groupBy(S.equals)('aabccd'.split(''))
+  //. [['a', 'a'], ['b'], ['c', 'c'], ['d']]
+  //.
+  //. > S.groupBy(S.equals)([1, 1, 2, 1, 1])
+  //. [[1, 1], [2], [1, 1]]
+  //.
+  //. > var P3 = S.curry2(function(a, b) { return a * b % 3 === 0; });
+  //. > S.groupBy(P3, [1, 2, 3, 4, 5, 6, 7, 8, 9])
+  //. [[1], [2, 3], [4], [5, 6], [7], [8, 9]]
+  //. ```
+  function groupBy(fn, xs) {
+    if (xs[0] === undefined) {
```

Comment:
What are we trying to ascertain here? If we're checking to see whether `xs` is `[]` let's use `xs.length === 0` instead.